ui/src/app/components/extend-initial-credits-modal.tsx

```diff
@@ -0,0 +1,122 @@
+import * as React from 'react';
+import { useState } from 'react';
+
+import { Profile } from 'generated/fetch';
+
+import { Button, StyledExternalLink } from 'app/components/buttons';
+import {
+  Modal,
+  ModalBody,
+  ModalFooter,
+  ModalTitle,
+} from 'app/components/modals';
+import { Spinner } from 'app/components/spinners';
+import { profileApi } from 'app/services/swagger-fetch-clients';
+import colors from 'app/styles/colors';
+import { displayDateWithoutHours, minusDays } from 'app/utils/dates';
+import { notificationStore, serverConfigStore } from 'app/utils/stores';
+import { supportUrls } from 'app/utils/zendesk';
+
+import { AoU } from './text-wrappers';
+
+const ExtensionDescription = ({ expirationDate }) => {
+  const {
+    initialCreditsValidityPeriodDays,
+    initialCreditsExpirationWarningDays,
+  } = serverConfigStore.get().config;
+
+  const eligibilityDate = displayDateWithoutHours(
+    minusDays(expirationDate, initialCreditsExpirationWarningDays)
+  );
+
+  return (
+    <div>
+      Initial credits provided by the <AoU /> Research Program expire after{' '}
+      {initialCreditsValidityPeriodDays} days. You are eligible to request an
+      extension after {eligibilityDate} which is{' '}
```

Comment:
this text feels weird for the case where this date is in the past.  Maybe drop the dates in that case?

Reply:
This modal should only appear when that date is in the past. I rewrote the text to:
```
  Initial credits provided by the <AoU /> Research Program expire after
  {initialCreditsValidityPeriodDays} days. You are currently eligible to request an
  extension.
```